File: bitSetGo-v2.0/C-Tricky-Journey/logic.java

import java.util.*; 
import java.lang.*; 
import java.io.*; 

class TJ
{ 
	
  public static void printNSE(int arr[], int n) 
  { 
	Stack<Integer> s = new Stack<Integer>(); 
	HashMap<Integer,Integer> mp = new HashMap<Integer,Integer>(); 
	s.push(arr[0]); 
		
	for (int i = 1; i < n; i++) { 

		if (s.empty()) { 
			s.push(arr[i]); 
			continue; 
		} 
		
	
	    while (s.empty() == false && s.peek() > arr[i]) { 
			mp.put(s.peek(), arr[i]); 
			s.pop(); 
		} 
		
		s.push(arr[i]); 
	} 
		
		
	while (s.empty() == false) { 
		mp.put(s.peek(), -1); 
		s.pop(); 
	} 
	
	
	for (int i=0; i<n; i++) 
	System.out.println( mp.get(arr[i])); 
 }  
	
	
public static void main (String[] args)throws IOException 
{
		InputStreamReader read = new InputStreamReader(System.in);
		BufferedReader in = new BufferedReader(read);
	    int t,i,j,n;
	    System.out.println("Enter");
	    t = Integer.parseInt(in.readLine());
	    for(i=0;i<t;i++)
	    {
	        n = Integer.parseInt(in.readLine());
	        int arr[]=new int[n];
	        for(j=0;j<n;j++)
	        {
	            arr[j] = Integer.parseInt(in.readLine());
	        }
	        int l = arr.length;
	        printNSE(arr,l);
	    
	     } 
} 
}

File: bitSetGo-v2.0/C-Tricky-Journey/logic.cpp
#include <bits/stdc++.h>
using namespace std;

void printNSE(int arr[], int n)
{
    stack<int> s;
    unordered_map<int, int> mp;


    s.push(arr[0]);



    for (int i = 1; i < n; i++) {

        if (s.empty()) {
            s.push(arr[i]);
            continue;
        }

    smaller and stack is not empty */
        while (s.empty() == false && s.top() > arr[i]) {
            mp[s.top()] = arr[i];
            s.pop();
        }


        s.push(arr[i]);
    }

    while (s.empty() == false) {
         mp[s.top()] = -1;
        s.pop();
    }

    for (int i=0; i<n; i++)
       cout << mp[arr[i]] << endl;
}


int main()
{
    int t,n;
    cin>>t;
    for(int i=0;i<t;i++)
    {
      cin>>n;
      int arr[n];
      for(int j=0;j<n;j++)
      {
        cin>>arr[n];
      }
      int l = sizeof(arr)/sizeof(arr[0]);
      printNSE(arr,l);
    }

    return 0;
}
